src/main/java/net/helpscout/api/ApiClient.java

```diff
@@ -1452,61 +1450,46 @@ private void setThreadProperties(ConversationThread thread) {
     }
     
     private <T> T doPost(String url, String requestBody, int expectedCode, ResultExtractor<T> extractor) throws ApiException {
-        HttpURLConnection conn = null;
-        try {
-            conn = HTTPConnectionUtils.getConnection(apiKey, baseUrl + url, METHOD_POST, expectedCode, requestBody);
+
+        try (HTTPConnectionWrapper conn = new HTTPConnectionWrapper(apiKey, baseUrl + url, METHOD_POST, expectedCode, requestBody)) {
             return extractor.extract(conn);
         } catch(ApiException ex) {
             throw ex;
         } catch (Exception ex) {
             throw new RuntimeException(ex);
-        } finally {
-            HTTPConnectionUtils.close(conn);
         }
     }
 
     private void doPut(String url, String requestBody, int expectedCode) throws ApiException {
-        HttpURLConnection conn = null;
-        try {
-            conn = HTTPConnectionUtils.getConnection(apiKey, baseUrl + url, METHOD_PUT, expectedCode, requestBody);
+
+        try (HTTPConnectionWrapper conn = new HTTPConnectionWrapper(apiKey, baseUrl + url, METHOD_PUT, expectedCode, requestBody)) {
         } catch(ApiException ex) {
             throw ex;
         } catch (Exception ex) {
             ex.printStackTrace();
             throw new RuntimeException(ex);
-        } finally {
-            HTTPConnectionUtils.close(conn);
         }
     }
 
     private String doGet(String url, int expectedCode) throws ApiException {
-        HttpURLConnection conn = null;
         String response    = null;
-        try {
-            conn = HTTPConnectionUtils.getConnection(apiKey, baseUrl + url, METHOD_GET, expectedCode);
-            response = HTTPConnectionUtils.getResponse(conn);
-
+        try (HTTPConnectionWrapper conn = new HTTPConnectionWrapper(apiKey, baseUrl + url, METHOD_GET, expectedCode)) {
```

Comment:
What is the profit of one more Utils class, which just will be in fact one line factory for Wrapper object?